lite/valor_lite/classification/computation.py

```diff
@@ -0,0 +1,330 @@
+import numpy as np
+from numpy.typing import NDArray
+
+# 0 gt label index
+# 1 pd label index
+# 2 score
+
+
+def compute_metrics(
+    data: NDArray[np.floating],
+    label_metadata: NDArray[np.int32],
+    score_thresholds: NDArray[np.floating],
+    n_datums: int,
+) -> tuple[
+    NDArray[np.int32],
+    NDArray[np.floating],
+    NDArray[np.floating],
+    NDArray[np.floating],
+    NDArray[np.floating],
+    NDArray[np.floating],
+    NDArray[np.floating],
+]:
+    """
+    Computes classification metrics.
+
+    Takes data with form:
+
+    Index 0 - GroundTruth Label Index
+    Index 1 - Prediction Label Index
+    Index 2 - Score
+
+    Parameters
+    ----------
+    data : NDArray[np.floating]
+        A sorted array of classification pairs.
+    label_metadata : NDArray[np.int32]
+        An array containing metadata related to labels.
+    score_thresholds : NDArray[np.floating]
+        An array contains score thresholds to compute metrics over.
+
+    Returns
+    -------
+    NDArray[np.int32]
+        TP, FP, FN, TN counts.
+    NDArray[np.floating]
+        Precision.
+    NDArray[np.floating]
+        Recall.
+    NDArray[np.floating]
+        Accuracy
+    NDArray[np.floating]
+        F1 Score
+    NDArray[np.floating]
+        ROCAUC.
+    NDArray[np.floating]
+        mROCAUC.
+    """
+
+    n_labels = label_metadata.shape[0]
+    n_label_keys = np.unique(label_metadata[:, 2]).size
+    n_scores = score_thresholds.shape[0]
+
+    pd_labels = data[:, 1].astype(int)
+
+    mask_matching_labels = np.isclose(data[:, 0], data[:, 1])
+    mask_score_nonzero = ~np.isclose(data[:, 2], 0.0)
+
+    # ==== calculate ROCAUC ====
```

Comment:
instead of adding these comments into one long function, could you split this function's contents into smaller functions (e.g., `_calculate_rocauc`)?